### DISCUSSION TITLE
Synthetic italic font not generated

### DISCUSSION THREAD

**Discussion by polirritmico** [ghostty-org/ghostty#3588] [Q&A] [answered]
Hi, great project.

From what I've read on the docs the `font-synthetic-style` should be enabled by default and it should work without user intervention if the font set in `font-family` doesn't have a stylistic variant (italic in my case), but I can't get the italic font generation working with my custom font.

My full Ghostty config:

```
font-size = 15
font-family = FiraCode Nerd Font

bold-is-bright = true
cursor-style = block
cursor-style-blink = false
gtk-titlebar = false
theme = Linux

keybind = ctrl+shift+semicolon=reload_config
```

My system info:

```
Ghostty 1.0.1-main+8111f5b9

Version
  - version: 1.0.1-main+8111f5b9
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.14.4
    runtime    : 4.14.4
  - libadwaita : enabled
    build      : 1.5.3
    runtime    : 1.5.3
```

```
Operating System: Gentoo Linux 2.17
KDE Plasma Version: 6.2.4
KDE Frameworks Version: 6.7.0
Qt Version: 6.8.1
Kernel Version: 6.6.67-gentoo (64-bit)
Graphics Platform: Wayland
```

```
dev-lang/zig-0.13.0-r2:0.13::gentoo  USE="llvm -doc -verify-sig" LLVM_SLOT="18"
gui-libs/gtk-4.14.4-r1:4::gentoo  USE="X cups introspection wayland (-aqua) -broadway -cloudproviders -colord -examples -gstreamer -sysprof -test (-vulkan)" CPU_FLAGS_X86="f16c"
gui-libs/libadwaita-1.5.3:1::gentoo  USE="introspection vala -test"
media-libs/harfbuzz-10.1.0:0/6.0.0::gentoo  USE="cairo glib graphite icu introspection truetype -debug -doc -experimental -test" ABI_X86="32 (64) (-x32)"
media-libs/freetype-2.13.3:2::gentoo  USE="X adobe-cff brotli bzip2 cleartype-hinting harfbuzz png svg -debug -doc -fontforge -static-libs -utils" ABI_X86="32 (64) (-x32)"
```

Compiled locally with: `zig build -p $HOME/.local -Doptimize=ReleaseFast`

---

At first I though it could be that the font is in a symlink inside `.local/share/fonts/`, but Ghostty list the font and uses the regular without problems (for testing purposes I've also copied it on the path with the same results):

```
$ ghostty +list-fonts | grep FiraCode
FiraCode Nerd Font
  FiraCode Nerd Font Bold
  FiraCode Nerd Font Medium Regular
```

From the logs I get this info:

```
info(font_shared_grid_set): font regular: FiraCode Nerd Font Medium Regular
info(font_shared_grid_set): font bold: FiraCode Nerd Font Bold
info(font_shared_grid_set): font italic: FiraCode Nerd Font Medium Regular
info(font_shared_grid_set): font bold_italic: FiraCode Nerd Font Bold
```

And this output:

![Ghostty-output](https://github.com/user-attachments/assets/155b0150-2e1d-4b6a-988f-8f4a479e2426)

Same font in Konsole:

![Konsole-output](https://github.com/user-attachments/assets/b2534616-553b-4bbc-91fb-8ceb68c87866)

Also, I've tried with:

```
font-family = ""
font-family = FiraCode Nerd Font
font-synthetic-style = true
```

Same results.

And if I set something like this:

```
font-size = 15
font-family = ""
font-family = FiraCode Nerd Font
font-family-italic = FireCode Nerd Font
font-synthetic-style = true
```

Then a different fonts is used for the italics. From the logs:

```
info(font_shared_grid_set): font regular: FiraCode Nerd Font Medium Regular
info(font_shared_grid_set): font bold: FiraCode Nerd Font Bold
info(font_shared_grid_set): font italic: Liberation Mono Italic
info(font_shared_grid_set): font bold_italic: FiraCode Nerd Font Bold
```

If relevant, at the top of the log, the **harfbuzz 8.4.0** caught my attention. Do I need to install that version on my system?:

```
info: ghostty version=1.0.1-main+8111f5b9
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.gtk
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=8.4.0
info: dependency fontconfig=21402
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
info(os): setlocale from env result=LC_CTYPE=es_CL.utf8;LC_NUMERIC=es_CL.utf8;LC_TIME=es_ES.UTF-8;LC_COLLATE=es_CL.utf8;LC_MONETARY=es_CL.UTF-8;LC_MESSAGES=es_CL.utf8;LC_PAPER=es_CL.utf8;LC_NAME=es_CL.utf8;LC_ADDRESS=es_CL.utf8;LC_TELEPHONE=es_CL.utf8;LC_MEASUREMENT=es_CL.utf8;LC_IDENTIFICATION=es_CL.utf8
info(gtk): GTK version build=4.14.4 runtime=4.14.4
info: reading configuration file path=/home/eduardo/.config/ghostty/config
info(config): default shell source=env value=/bin/bash
info(gtk): libadwaita version build=1.5.3 runtime=1.5.3
```

In case is useful this is my custom font: [FiraCodeNerdFont-Regular.zip](https://github.com/user-attachments/files/18263948/FiraCodeNerdFont-Regular.zip)

Don't know if more info is relevant.

Any tip or help would be appreciated. Thanks!

**Comment by polirritmico** [comment]
Hi! Since Gentoo now has an official ebuild for Ghostty, I've remove my local compilation, and use the official one. Now is using my system libraries, I've compile it successfully with `adwaita + gtk` enabled and also with `glfw` but still the same results.

Any idea or tip on how to debug this?

```
❯ ghostty
info: ghostty version=1.0.0
info: ghostty build optimize=ReleaseFast
info: runtime=apprt.Runtime.glfw
info: font_backend=font.main.Backend.fontconfig_freetype
info: dependency harfbuzz=10.1.0
info: dependency fontconfig=21500
info: renderer=renderer.OpenGL
info: libxev backend=main.Backend.io_uring
```

---

Edit: I've just tried with the non-nerdfont/non-custom-patched version of [FiraCode](https://github.com/tonsky/FiraCode/releases/tag/6.2) and got the same issue, no synthesized italic.

**Reply by rhodes-b** [reply]
btw dont use the glfw runtime its meant for developer use only and is missing tons and tons of features / has known issues that will not get fixed because its for development only

**Reply by polirritmico** [reply]
Yes, I just test them both to discard a library issue. Anyway, thanks for the reply.

The problem persists with 1.1.0. Don't know why this isn't considered a bug.

**Reply by rhodes-b** [reply]
Id say it is as a few others have also posts in the discord. Your post was pretty close after launch so it got seemingly got lost in the wave of many posts. 

Im not the bdfl so i can't make this an issue

**Comment by 00-kat** [comment, marked as answer]
#3791 is a duplicate of this, but all the conversation seems to be in #3791, so I think closing this in favour of #3791 would be warranted.  If there's any information in this thread that's not already covered in #3791, it would be nice if someone (preferably the OP) were to move it across ^^.

**Reply by 00-kat** [reply]
Oh my, I just realised this was about italics while the other one was about bold, I'll add a note there mentioning that the issue is with font synthesis in general and not specifically bold.

**Reply by polirritmico** [reply]
> it would be nice if someone (preferably the OP) were to move it across ^^

Done 👍
